polkadot/runtime/src/lib.rs

```diff
@@ -14,84 +14,286 @@
+extern crate substrate_runtime_consensus as consensus;
+extern crate substrate_runtime_council as council;
+extern crate substrate_runtime_democracy as democracy;
+extern crate substrate_runtime_executive as executive;
+extern crate substrate_runtime_session as session;
+extern crate substrate_runtime_staking as staking;
+extern crate substrate_runtime_system as system;
+extern crate substrate_runtime_timestamp as timestamp;
 extern crate polkadot_primitives;
 
-#[cfg(all(feature = "std", test))]
-extern crate substrate_keyring as keyring;
+mod parachains;
 
-#[cfg(feature = "std")]
-extern crate rustc_hex;
+use rstd::prelude::*;
+use runtime_io::BlakeTwo256;
+use polkadot_primitives::{AccountId, Balance, BlockNumber, Hash, Index, SessionKey, Signature};
+use runtime_primitives::generic;
+use runtime_primitives::traits::{Identity, HasPublicAux};
+pub use runtime_primitives::BuildExternalities;
 
-#[cfg_attr(any(test, feature = "std"), macro_use)]
-extern crate substrate_primitives as primitives;
+pub struct Concrete;
 
-#[macro_use]
-extern crate substrate_runtime_io as runtime_io;
+impl HasPublicAux for Concrete {
+	type PublicAux = AccountId;	// TODO: Option<AccountId>
+}
+
+impl system::Trait for Concrete {
+	type Index = Index;
+	type BlockNumber = BlockNumber;
+	type Hash = Hash;
+	type Hashing = BlakeTwo256;
+	type Digest = generic::Digest<Vec<u8>>;
+	type AccountId = AccountId;
+	type Header = generic::Header<BlockNumber, Hash, Vec<u8>>;
+}
+pub type System = system::Module<Concrete>;
+
+impl consensus::Trait for Concrete {
+	type PublicAux = <Self as HasPublicAux>::PublicAux;
+	type SessionKey = SessionKey;
+}
+pub type Consensus = consensus::Module<Concrete>;
+pub use consensus::Call as ConsensusCall;
+
+impl timestamp::Trait for Concrete {
+	type Value = u64;
+}
+pub type Timestamp = timestamp::Module<Concrete>;
+pub use timestamp::Call as TimestampCall;
+
+impl session::Trait for Concrete {
+	type ConvertAccountIdToSessionKey = Identity;
+}
+pub type Session = session::Module<Concrete>;
+
+impl staking::Trait for Concrete {
+	type Balance = Balance;
+	type DetermineContractAddress = BlakeTwo256;
+}
+pub type Staking = staking::Module<Concrete>;
+
+impl democracy::Trait for Concrete {
+	type Proposal = PrivCall;
+}
+pub type Democracy = democracy::Module<Concrete>;
 
+impl council::Trait for Concrete {}
+pub type Council = council::Module<Concrete>;
+pub type CouncilVoting = council::voting::Module<Concrete>;
+
+impl parachains::Trait for Concrete {}
+pub type Parachains = parachains::Module<Concrete>;
+
+impl_outer_dispatch! {
+	pub enum Call where aux: <Concrete as HasPublicAux>::PublicAux {
+		Consensus = 0,
+		Session = 1,
+		Staking = 2,
+		Timestamp = 3,
+		Democracy = 5,
+		Council = 6,
+		CouncilVoting = 7,
+	}
+
+	pub enum PrivCall {
+		Consensus = 0,
+		Session = 1,
+		Staking = 2,
+		Democracy = 5,
+		Council = 6,
+		CouncilVoting = 7,
+	}
+}
+
+pub type Header = generic::Header<BlockNumber, Hash, Vec<u8>>;
```

Comment:
seems like changes from the end of the last PR were lost

Reply:
may have got clobbered. my god i hate that git can't support merging squashed commits.